File: tyler/a2/readability.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    string text = get_string("Text:");
    
    int i = 0;
    int words = 0;
    int sentences = 0;
    int letters = 0;
    do
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        i++;
    }
    while (text[i] != '\0');
    words++;
    
    float L = (float) letters/ (float) words * 100.0;
    float S = (float) sentences/ (float) words * 100.0;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    
    int grade = (int) index;
    if (index - (int) index > 0.5)
    {
        grade++;
    }
    
    if (grade < 1)
    {
        grade = 1;
        printf("Grade %d\n", grade);
    }
    else if (grade > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

File: tyler/a6/readability/readability.py
from cs50 import get_string

s = get_string("Text: ")
letters   = sum(c.isalpha() for c in s)
words  = (sum(c.isspace() for c in s)+1)/100
sentances = s.count(".")+s.count("!")+s.count("?")
CL = 0.0588 * letters/words - 0.296 * sentances/words - 15.8
# print(CL)
if CL >= 16:
    print("Grade 16+")
elif CL < 1:
    print("Before Grade 1")
else:
    print(f"Grage {int(round(CL,0))}")